### PR TITLE
Replacing hardcoded namespace in pilot helm template

### PR DESCRIPTION
**Pull request by depohmel**
with original template installing into custom ns
was failing with following error
```
helm install install/kubernetes/helm/istio --name istio-test \
                                           --namespace test-namespace \
                                           --set global.oneNamespace=true \
                                           --set sidecarInjectorWebhook.enabled=false \
                                           --set gateways.enabled=false
Error: release istio-test failed: namespaces "istio-system" not found
```

With this fix error is gone and pilot is configurated with proper ns
```
helm install ~/istio/install/kubernetes/helm/istio --name istio-test \
                                                   --namespace test-namespace \
                                                   --set global.oneNamespace=true \
                                                   --set sidecarInjectorWebhook.enabled=false \
                                                   --set gateways.enabled=false
NAME:   istio-test
LAST DEPLOYED: Wed Aug 22 13:22:59 2018
NAMESPACE: test-namespace

```

### CODE REVIEW

**Comment by depohmel** [comment]
I signed it!

**Comment by depohmel** [comment]
I signed it! 
and it got accepted by Google :)

File: install/kubernetes/helm/istio/charts/pilot/templates/gateway.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: networking.istio.io/v1alpha3
 kind: Gateway
 metadata:
   name: istio-autogenerated-k8s-ingress
-  namespace: istio-system
+  namespace: {{ default .Release.Namespace .Values.global.istioNamespace }}
```
Comment (sdake) [inline install/kubernetes/helm/istio/charts/pilot/templates/gateway.yaml]: namespace: {{ .Release.Namespace }}

File: install/kubernetes/helm/istio/charts/pilot/templates/meshexpansion.yaml [hunk 2]
```diff
@@ -1,20 +1,20 @@
 {{- if .Values.global.meshExpansion }}
-
+{{ $istioNamespace := default .Chart.Namespace .Values.global.istioNamespace }}
```
Comment (sdake) [inline install/kubernetes/helm/istio/charts/pilot/templates/meshexpansion.yaml]: there is no definition in `values.yaml` of `global.istioNamespace`.  Hence this seems redundant.

Comment (depohmel) [reply install/kubernetes/helm/istio/charts/pilot/templates/meshexpansion.yaml]: @sdake, thanks for the review.
I see it used in gateways chart:
```
./install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml:        {{- if $.Values.global.istioNamespace }}
./install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml:          - zipkin.{{ $.Values.global.istioNamespace }}:9411
./install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml:          {{- if $.Values.global.istioNamespace }}
./install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml:          - istio-pilot.{{ $.Values.global.istioNamespace }}:15011
./install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml:          {{- if $.Values.global.istioNamespace }}
./install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml:          - istio-pilot.{{ $.Values.global.istioNamespace }}:15010
```
And `global` is a specialcase in Helm as I recall. But sure thing I can remove it.

**Comment by depohmel** [comment]
@sdake, done

**Comment by depohmel** [comment]
@sdake, could you please take another look?

File: install/kubernetes/helm/istio/charts/pilot/templates/meshexpansion.yaml (hunk 2 above)
Comment (sdake) [reply install/kubernetes/helm/istio/charts/pilot/templates/meshexpansion.yaml]: hmm, that variable is not present in `values.yaml` - maybe a bug in the gateways chart.  I'm not sure how this works :)

Comment (depohmel) [reply install/kubernetes/helm/istio/charts/pilot/templates/meshexpansion.yaml]: As I recall (I might be wrong) helm is gathering Values.global from all charts main and subs, combines them and then it's available for all charts. So you don't have to declare it in values.yaml, you can just use it :) 
Another thing, you don't have to declare a value in values.yaml to use it. Once you run helm with --set or -f, these specified values will be added into chart .Values and available to use.

**Comment by sdake** [comment]
@depohmel not sure where `--set global.oneNamespace=true` comes from, but this PR looks gtg.

/lgtm
/approve

**Pull request merged**